src/nemo_run/core/execution/docker.py

```diff
@@ -181,25 +181,29 @@ def package_configs(self, *cfgs: tuple[str, str]) -> list[str]:
 
     def package(self, packager: Packager, job_name: str):
         assert self.experiment_id, "Executor not assigned to an experiment."
-        output = subprocess.run(
-            ["git", "rev-parse", "--show-toplevel"],
-            check=True,
-            stdout=subprocess.PIPE,
-        )
-        path = output.stdout.splitlines()[0].decode()
-        base_path = Path(path).absolute()
+        if isinstance(packager, GitArchivePackager):
+            output = subprocess.run(
+                ["git", "rev-parse", "--show-toplevel"],
+                check=True,
+                stdout=subprocess.PIPE,
+            )
+            path = output.stdout.splitlines()[0].decode()
+            base_path = Path(path).absolute()
+        else:
+            base_path = Path(os.getcwd()).absolute()
         local_pkg = packager.package(base_path, self.job_dir, job_name)
         local_code_extraction_path = os.path.join(self.job_dir, "code")
         ctx = Context()
-        ctx.run(f"mkdir -p {local_code_extraction_path}")
+        if local_pkg:
+            ctx.run(f"mkdir -p {local_code_extraction_path}")
```

Comment:
For now I think you need to run this at all times due to https://github.com/NVIDIA/NeMo-Run/blob/8e3446d95d5806989c0bb7cc65e41a2fe1077adb/src/nemo_run/core/execution/docker.py#L277